lib/pitchfork/chunked.rb

```diff
@@ -106,6 +106,7 @@ def call(env)
 
       if !env.key?('rack.hijack_io') && # full highjack
          !headers['rack.hijack'] && # partial hijack
+         !body.respond_to?(:call) && # streaming body
```

Comment:
I'm not completely sure this is correct, but it's Friday so... I'm not operating at 100% right now.

I think what we want here is `body.respond_to?(:each)`.

We actually had a bit of a long discussion about this, but there are some middleware that do this:

```
class HelloWorldMiddleware
  def initialise(app)
    @app = app
  end

  def each
    yield "Hello World"
  end

  def call(env)
    return [200, {}, self]
  end
end
```

In the above case, `body.respond_to?(:call)` is true, but it's not a streaming body, because it's also `body.respond_to(:each)`. We had to have this logic to avoid compatibility issues for the above pattern, even though we strongly advise middleware doesn't return itself (or `self.dup`) as the body.